### PR TITLE
sync our copy of droplet with whats in the droplet/code-dot-org branch

### PR DESCRIPTION
**Pull request by davidsbailey**
I've inspected the contents of droplet-full.js to confirm that it contains evidence of the latest changes from https://github.com/droplet-editor/droplet/tree/code-dot-org , however when I rebuild droplet and copy it over in this PR there are substantial diffs. Since I'm going to be making a bunch of droplet changes, I want to merge these diffs into our repo now to isolate any problems they may cause from any problems that may be caused by future substantial changes.

Here's what I did to update these files:
```
Davids-MacBook-Pro-2:~/src/droplet (code-dot-org)$ git branch 
* code-dot-org
  master
Davids-MacBook-Pro-2:~/src/droplet (code-dot-org)$ git pull
Already up-to-date.
Davids-MacBook-Pro-2:~/src/droplet (code-dot-org)$ npm install
Davids-MacBook-Pro-2:~/src/droplet (code-dot-org)$ LANGUAGE=javascript grunt dist ; cp dist/* ~/src/cdo/apps/lib/droplet/
Running "browserify:build" (browserify) task
>> Bundle dist/droplet-full.js created.

Running "uglify:build" (uglify) task
>> 1 sourcemap created.
>> 1 file created 642.42 kB → 311.1 kB

Running "cssmin:minify" (cssmin) task
>> 1 file created. 11.1 kB → 9.47 kB

Done, without errors.
Davids-MacBook-Pro-2:~/src/droplet (code-dot-org)$ 
```

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
I was also unable to generate an identical build. I wonder if we should start some best practices around this for when we create new droplet builds.

Some info it would probably be useful to include every time we generate a new droplet that has these sorts of noisy changes (i.e. changes that seem unrelated to any code changes we made in droplet)

1) Version of `node` used to generate build.
2) Results of `npm ls` in droplet repo? (This will be rather long, but I think useful in tracking down what might have changed).
3) Can you think of anything else worthwhile?

**Comment by davidsbailey** [comment]
If we're including the `node` version then we should probably include the `npm` version as well.

**Comment by Bjvanminnen** [comment]
I think that's fine to include, but theoretically it should be superfluous. Thinking about this more, I think even node version shouldn't really matter, and that two different versions of node using the same set of node_modules should produce identical builds.

tldr: Let's just always include node version, npm version, npm ls (the last of which might be most important).

**Comment by davidsbailey** [comment]
```
Davids-MacBook-Pro-2:~/src/droplet (code-dot-org)$ node --version
v6.9.1
Davids-MacBook-Pro-2:~/src/droplet (code-dot-org)$ npm --version
3.10.8
```
output from `npm ls`: 
[npm-ls.txt](https://github.com/code-dot-org/code-dot-org/files/1075558/npm-ls.txt)

**Comment by davidsbailey** [comment]
It's possible that pinning versions of packages we depend on is insufficient given that those packages do not always depend on specific versions of their dependencies. For example node_modules/events/package.json contains the following:
```
  "devDependencies": {
    "mocha": "~1.21.4",
    "zuul": "~1.10.2"
  },
```

**Comment by Bjvanminnen** [comment]
That's a good point (though I'd done a build using my old state, including the old state of my node_modules and was still seeing significant differences).

I wonder if we should consider using yarn, which I think solves this problem.
